.github/workflows/buildpack-deps-emscripten.yml

```diff
@@ -0,0 +1,34 @@
+name: buildpack-deps:emscripten
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/buildpack-deps-emscripten.yml'
+      - 'scripts/docker/buildpack-deps/Dockerfile.emscripten'
+
+jobs:
+  build:
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      DOCKER_REPOSITORY: docker.pkg.github.com/${{ github.repository }}
+      IMAGE_NAME: buildpack-deps
+      IMAGE_VARIANT: emscripten
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Upgrade ${{ env.IMAGE_NAME }}-${{ env.IMAGE_VARIANT }}
+        run: |
+          echo "${GITHUB_TOKEN}" | docker login docker.pkg.github.com -u "${GITHUB_ACTOR}" --password-stdin
+          scripts/ci/docker_upgrade.sh
```

Comment:
Is there any particular reason not to pass `$DOCKER_REPOSITORY` and the other variables via script parameters? I just realized that it must be using these variables but it's not all that obvious at first. And when you see only the .yml file you have no way to see which script is using them. They're not even exported (does github just automatically export all the env variables?).

I think that passing them via parameters would be much more straightforward.

Reply:
Thats true, its more clean.